### ISSUE TITLE
npm ERR! Darwin 13.4.0

### ISSUE THREAD

**Issue by mattia-pink** [npm/npm#7555]
I have this output in the terminal when I run this: npm install

```
npm WARN engineStrict Per-package engineStrict (found in package.json for gulp-concat)
npm WARN engineStrict won't be used in npm 3+. Use the config setting `engine-strict` instead.
npm WARN engineStrict Per-package engineStrict (found in package.json for gulp)
npm WARN engineStrict won't be used in npm 3+. Use the config setting `engine-strict` instead.
npm WARN engineStrict Per-package engineStrict (found in package.json for vinyl-fs)
npm WARN engineStrict won't be used in npm 3+. Use the config setting `engine-strict` instead.
npm ERR! Darwin 13.4.0
npm ERR! argv "node" "/usr/local/bin/npm" "install"
npm ERR! node v0.12.0
npm ERR! npm  v2.7.0

npm ERR! null == true
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <http://github.com/npm/npm/issues>
```
how can I fix this issue? Any idea on this?

**Comment by cibulka** [comment]
Same here!

```
grunt-newer@1.1.0 node_modules/grunt-newer
├── rimraf@2.2.8
└── async@0.9.0
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/Users/Cibulka/.node/bin/npm" "update"
npm ERR! node v0.12.0
npm ERR! npm  v2.7.0

npm ERR! null == true
npm ERR! 
```

**Comment by othiym23** [comment]
Could one of you upload a copy of `npm-debug.log` to gist.gitub.com? There's either a bug in npm or a bug in node that's causing this to happen, but without a stack trace to look at it's impossible to figure out what's going on. Thanks a lot in advance for your help!

**Comment by mattia-pink** [comment]
here you are https://gist.github.com/mattia-pink/e2fa912ad084decee5d7

thanks

**Comment by cibulka** [comment]
Sure, thanks for quick reply! https://gist.github.com/cibulka/24b5597286a76d232746

**Comment by othiym23** [comment]
Yay! (Not yay for the problems, obviously, but those debug logs were _very_ helpful!) Thanks so much! See #7349 for further discussion of this, as I'm going to close this issue and use that one as the master issue for tracking this problem, which appears to be in Node.js's HTTPS handling code.

**Comment by cibulka** [comment]
Yay for your superquick responses! Will follow #7349 for more.

**Comment by TommyHili** [comment]
I need help, my computer have error 
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "update" "-g" "npm"
npm ERR! node v0.12.1
npm ERR! npm  v2.5.1
npm ERR! path /usr/local/lib/node_modules/npm
npm ERR! code EACCES
npm ERR! errno -13

npm ERR! Error: EACCES, unlink '/usr/local/lib/node_modules/npm'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES, unlink '/usr/local/lib/node_modules/npm']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   path: '/usr/local/lib/node_modules/npm' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.
npm ERR! error rolling back Error: EACCES, unlink '/usr/local/lib/node_modules/npm'
npm ERR! error rolling back     at Error (native)
npm ERR! error rolling back  { [Error: EACCES, unlink '/usr/local/lib/node_modules/npm']
npm ERR! error rolling back   errno: -13,
npm ERR! error rolling back   code: 'EACCES',
npm ERR! error rolling back   path: '/usr/local/lib/node_modules/npm' }

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/LEE/npm-debug.log

**Comment by KenanY** [comment]
@TommyHili Your issue seems completely unrelated to this issue. Regardless, it looks like you need to prefix the command you are running with `sudo`, since you seem to not have the permissions necessary to write to `/usr/local`.

**Comment by TommyHili** [comment]
@KenanY sorry that is my foot,I just update Node to 0.12.1, next time Similar problems to tack overflow ,think you Answer,

**Comment by camilodelvasto** [comment]
Any word on this? I'm having the same issue:
Node is downgraded to v0.10.40
npm is version 2.12.1
```
[sqlite3] Success: "/usr/local/lib/node_modules/jsbin/node_modules/sqlite3/lib/binding/node-v11-darwin-x64/node_sqlite3.node" is installed via remote
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "i" "-g" "jsbin"
npm ERR! node v0.10.40
npm ERR! npm  v2.12.1
npm ERR! path /Users/cdelvasto/.npm/connect/2.7.2
npm ERR! code EACCES
npm ERR! errno 3

npm ERR! Error: EACCES, mkdir '/Users/cdelvasto/.npm/connect/2.7.2'
npm ERR!  { [Error: EACCES, mkdir '/Users/cdelvasto/.npm/connect/2.7.2']
npm ERR!   errno: 3,
npm ERR!   code: 'EACCES',
npm ERR!   path: '/Users/cdelvasto/.npm/connect/2.7.2',
npm ERR!   parent: 'express' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/cdelvasto/Documents/programming/jsbin/npm-debug.log
npm i -g jsbin  14.29s user 5.90s system 15% cpu 2:12.20 total
```

**Comment by othiym23** [comment]
@camilodelvasto Your issue is unrelated to the original post, but I believe it can be fixed by running `sudo chown -R $(whoami) "$HOME/.npm"`, after first making sure that you've upgraded to the latest stable version of npm (`npm -g install npm`).

**Comment by camilodelvasto** [comment]
@othiym23 that did the trick, thanks!

**Comment by NicolasErices** [comment]
othiym23 gracias me  funciono tenia el mismo problema que @camilodelvasto

**Comment by yanshinian** [comment]
You can reinstall node and npm.(https://nodejs.org/en/download/).That's what I did.

**Comment by brody182** [comment]
install the stable version of node.js

**Comment by davecurrierseo** [comment]
@othiym23 Thank you!

**Comment by dkmin** [comment]
@othiym23 Thank you!
> sudo chown -R $(whoami) "$HOME/.npm"
> npm -g install npm

**Comment by jacoobs** [comment]
change de version npm with nvm ..
https://www.youtube.com/watch?v=SU8K8jj-fQs

**Comment by BrianBravo** [comment]
@jacoobs that vid is private :/

**Comment by maddymaster** [comment]
I get this error, any suggestions on how to solve it?
```
maddyspc:node_modules MaddyVeena$ sudo npm install angular-cli@latest -g
npm ERR! Darwin 15.2.0
npm ERR! argv "node" "/Users/MaddyVeena/.nvm/v0.9.6/bin/npm" "install" "angular-cli@latest" "-g"
npm ERR! node v0.9.6
npm ERR! npm  v3.10.3

npm ERR! Unknown encoding: buffer
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

buffer.js:405
      throw new Error('Unknown encoding: ' + encoding);
            ^
Error: Unknown encoding: buffer
    at Buffer.write (buffer.js:405:13)
    at new Buffer (buffer.js:261:26)
    at WriteStream.Writable.write (_stream_writable.js:134:13)
    at WriteStreamAtomic._write (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/node_modules/fs-write-stream-atomic/index.js:64:37)
    at doWrite (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/node_modules/readable-stream/lib/_stream_writable.js:319:64)
    at writeOrBuffer (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/node_modules/readable-stream/lib/_stream_writable.js:308:5)
    at WriteStreamAtomic.Writable.write (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/node_modules/readable-stream/lib/_stream_writable.js:246:11)
    at WriteStreamAtomic.Writable.end (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/node_modules/readable-stream/lib/_stream_writable.js:456:51)
    at writeLogFile (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/lib/utils/error-handler.js:205:8)
    at exit (/Users/MaddyVeena/.nvm/v0.9.6/lib/node_modules/npm/lib/utils/error-handler.js:82:5)
```

**Comment by KenanY** [comment]
@maddymaster Don't use Node.js v0.9.

**Comment by maddymaster** [comment]
this is resolved, I updated the node version to latest v6.7.0

**Comment by ndiyansah** [comment]
i want to update NPM ver 3 to 5 but cant  update, any suggestions on how to solve it?


```

npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/8.2.1/bin/node" "/Users/handi/n/bin/npm" "-g" "install" "npm"
npm ERR! node v8.2.1
npm ERR! npm  v3.10.10
npm ERR! path /Users/handi/.npm-packages/lib/node_modules/npm
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall access

npm ERR! enoent ENOENT: no such file or directory, access '/Users/handi/.npm-packages/lib/node_modules/npm'
npm ERR! enoent ENOENT: no such file or directory, access '/Users/handi/.npm-packages/lib/node_modules/npm'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/handi/npm-debug.log
```
